page_shield.go

```diff
@@ -0,0 +1,138 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+var ErrMissingScriptID = errors.New("required script id missing")
+
+type PageShieldScript struct {
+	ScriptID                string                     `json:"script_id,omitempty"`
+	ScriptURL               string                     `json:"script_url,omitempty"`
+	AddedAt                 *time.Time                 `json:"added_at,omitempty"`
+	FirstSeenAt             *time.Time                 `json:"first_seen_at,omitempty"`
+	LastSeenAt              *time.Time                 `json:"last_seen_at,omitempty"`
+	Host                    string                     `json:"host,omitempty"`
+	DomainReportedMalicious bool                       `json:"domain_reported_malicious,omitempty"`
+	SeenOnFirst             string                     `json:"seen_on_first,omitempty"`
+	SeenOn                  []string                   `json:"seen_on,omitempty"`
+	AppearsInCDNCGIPath     bool                       `json:"appears_in_cdn_cgi_path,omitempty"`
+	Hash                    string                     `json:"hash,omitempty"`
+	JsIntegrityScore        int                        `json:"js_integrity_score,omitempty"`
+	FetchedAt               *time.Time                 `json:"fetched_at,omitempty"`
+	Versions                []PageShieldScriptVersions `json:"versions,omitempty"`
+}
+
+type PageShieldScriptVersions struct {
+	Hash             string     `json:"hash,omitempty"`
+	JsIntegrityScore int        `json:"js_integrity_score,omitempty"`
+	FetchedAt        *time.Time `json:"fetched_at,omitempty"`
+}
+
+type ListPageShieldParameters struct {
+	ScriptPage          string `url:"script_page,omitempty"`
+	OrderBy             string `url:"order_by,omitempty"`
+	Hosts               string `url:"hosts,omitempty"`
+	ExcludeCDNCGI       bool   `url:"exclude_cdn_cgi,omitempty"`
+	Status              string `url:"status,omitempty"`
+	ExcludeScriptURLS   string `url:"exclude_script_urls,omitempty"`
+	PrioritizeMalicious bool   `url:"prioritize_malicious,omitempty"`
+	ScriptURLs          string `url:"script_urls,omitempty"`
+	Direction           string `url:"direction,omitempty"`
+	ResultInfo
+}
+
+type ListPageShieldResponse struct {
+	Result     []PageShieldScript `json:"result"`
+	ResultInfo `json:"result_info,omitempty"`
+	Response
+}
+
+// GetPageShieldStatus Fetches the Page Shield status.
+//
+// API Reference: https://api.cloudflare.com/#page-shield-get-page-shield-status
+func (api *API) GetPageShieldStatus(ctx context.Context, rc *ResourceContainer) error {
+	if rc.Identifier == "" {
+		return ErrMissingZoneID
+	}
+	uri := fmt.Sprintf("/zones/%s/page_shield", rc.Identifier)
+
+	_, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
```

Comment:
i've got a feeling, the docs are broken for this as we don't have a body. i'll ping the team.